### PR TITLE
Theme R1 anchors and buttons

### PR DESCRIPTION
**Pull request by jamiepaul**
Initial implementation of R1 anchors, buttons and CTA components. This is a work in progress as there are some design implementation discussions that need to happen surrounding buttons w/ icons and the transition.

Points to discuss: 
* Variable and mixin separation
* Component variations - base vs theme

### CODE REVIEW

File: wp-content/themes/core/pcss/base/typography/_anchors.pcss [hunk 1]
```diff
@@ -28,8 +28,8 @@ a {
 
 .anchor,
 .t-content a {
-	border-bottom: 2px solid var(--color-primary);
 	color: var(--color-link);
+	font-weight: var(--font-weight-semibold);
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/base/typography/_anchors.pcss]: Note that when you handle global project anchor styles, you'll typically find you need to do a "reset" pass in situations where anchors might be used (image with an anchor, youtube/vimeo lazy implementation, etc.) ; )

File: wp-content/themes/core/pcss/components/_cta.pcss [hunk 2]
```diff
@@ -5,31 +5,69 @@
  * ----------------------------------------------------------------------------- */
 
 .c-cta,
-a.c-cta,
-.t-content .c-cta {
-	border-bottom: none;
-	color: var(--color-link);
+a.c-cta {
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/components/_cta.pcss]: Fwiw, I prefer to setup all button styles (including the "this is a text link" anchor variation, we should also chat with design about the two variations of anchor) as a `c-btn-` and not prescribe a specific action style to the cta on it's own as the cta component itself will very likely need to change on what action variation it uses based on context of implementation.

File: wp-content/themes/core/pcss/components/_cta.pcss [hunk 3]
```diff
@@ -5,31 +5,69 @@
...
+	border-bottom: 2px solid var(--color-submarine);
+	color: var(--color-mine-shaft);
 	display: inline-block;
-	font-size: var(--font-size-1);
-	font-weight: var(--font-weight-bold);
+	font-family: var(--font-family-calibre);
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/components/_cta.pcss]: font shorthand property makes sense here?

File: wp-content/themes/core/pcss/components/_cta.pcss [hunk 4]
```diff
@@ -5,31 +5,69 @@
...
+		border-color: var(--color-mine-shaft);
+		color: var(--color-black-pearl);
+	}
+}
+
+/* -------------------------------------------------------------------------
+ * CTA: Modifier - Icon
+ * ------------------------------------------------------------------------- */
+
+.c-cta--icon,
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/components/_cta.pcss]: let's talk through this tomorrow

File: wp-content/themes/core/pcss/components/button/_outline.pcss [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+		background-color: transparent;
+		border-color: var(--color-link-hover);
+		color: var(--color-link-hover);
+	}
+
+	&:active {
+		background-color: var(--color-desert-storm);
+		border: 1px solid var(--color-mine-shaft);
+		color: var(--color-mine-shaft);
+		outline: 0;
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/components/button/_outline.pcss]: Don't nix outline, good for a11y ;)

File: wp-content/themes/core/pcss/utilities/mixins/_buttons.pcss [hunk 6]
```diff
@@ -1,45 +1,82 @@
...
- *
+ * Button - Base
  * ----------------------------------------------------------------------------- */
 
-@define-mixin button-submit {
-	@mixin button-global;
-
-	background-color: var(--color-black);
-	padding: 1em;
+@define-mixin button-base {
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/utilities/mixins/_buttons.pcss]: Given we have a `button-global` might provide greater naming clarity to have this be `button`?

File: wp-content/themes/core/pcss/components/button/_outline.pcss [hunk 7]
```diff
@@ -0,0 +1,71 @@
+/* -----------------------------------------------------------------------------
+ *
+ * Button: Outline
+ *
+ * ----------------------------------------------------------------------------- */
+
+.c-btn-outline,
+a.c-btn-outline {
+	@mixin button-global;
+	@mixin button-base;
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/components/button/_outline.pcss]: Let's talk through this tomorrow, I could see a case being made for nixing a `button-base` mixin to give us greater, less complicated ability to pivot styles for any given button while keeping down some of the css.

File: wp-content/themes/core/pcss/components/button/_solid.pcss [hunk 8]
```diff
@@ -15,91 +15,58 @@
...
-.c-btn--sm,
-a.c-btn--sm {
-	font-size: var(--font-size-1);
-	padding: 16px 18px;
+.c-btn--icon,
+a.c-btn--icon {
 
-	@media (--viewport-medium) {
-		font-size: var(--font-size-1);
+	&:after {
```
Comment (ryanurban) [inline wp-content/themes/core/pcss/components/button/_solid.pcss]: Let's ask design how they would imagine longer text playing out for a button with an icon, I could see us needing to do a bit more of a robust implementation where the icon is absolutely positioned so it stays to the right and vertically centered.

File: wp-content/themes/core/pcss/base/typography/_anchors.pcss (hunk 1 above)
Comment (jamiepaul) [reply wp-content/themes/core/pcss/base/typography/_anchors.pcss]: This was my misunderstanding of the inline link style. Removed it!

File: wp-content/themes/core/pcss/components/_cta.pcss (hunk 2 above)
Comment (jamiepaul) [reply wp-content/themes/core/pcss/components/_cta.pcss]: Added as `c-btn-text` and switched the CTA back to its previous styles

File: wp-content/themes/core/pcss/components/button/_outline.pcss (hunk 7 above)
Comment (jamiepaul) [reply wp-content/themes/core/pcss/components/button/_outline.pcss]: Removed the base mixin and added mixins for `button-solid` and `button-outline`. I thought adding one for `button-text` was overkill.

**Comment by jamiepaul** [comment]
Closing PR after merging to feature R1 theme updates branch

**Pull request merged**